M/Mesa/build_tarballs.jl

```diff
@@ -3,41 +3,111 @@
 using BinaryBuilder, Pkg
 
 name = "Mesa"
-version = v"20.1.5"
+version = v"25.2.1"
 
 # Collection of sources required to complete build
 sources = [
-    ArchiveSource("https://archive.mesa3d.org/mesa-$version.tar.xz", "fac1861e6e0bf1aec893f8d86dbfb9d8a0f426ff06b05256df10e3ad7e02c69b"),
+    ArchiveSource("https://archive.mesa3d.org/mesa-$(version).tar.xz",
+                  "c124372189d35f48e049ee503029171c68962c580971cb86d968a6771c965ba4"),
 ]
 
 # Bash recipe for building across all platforms
 script = raw"""
-mkdir build
-cd build
-apk add py3-mako
-meson -D b_ndebug=true -D buildtype=release -D strip=true -D llvm=false ../mesa* --cross-file="${MESON_TARGET_TOOLCHAIN}"
-ninja -j${nproc}
-ninja install
-mv $prefix/bin/opengl32.dll $prefix/bin/opengl32sw.dll
-install_license ../mesa*/docs/license.html
+cd ${WORKSPACE}/srcdir/mesa*
+
+apk add glslang-dev py3-mako py3-yaml
+
+# `${bindir}/wayland_scanner` is a symbolic link that contains `..` path elements.
+# These are not properly normalized: They are normalized before expanding the symbolic link `/workspace/destdir`,
+# and this leads to a broken reference. We resolve the path manually.
+#
+# Since this symbolic link is working in the shell, and since `pkg-config` outputs the expected values,
+# I think this may be a bug in `meson`.
+#
+# The file `wayland-scanner.pc` is mounted multiple times (and is also available via symbolic links).
```

Comment:
I'm confused by why you need all of this. With https://github.com/JuliaPackaging/BinaryBuilder.jl/pull/1346 the pkgconfig files should be relocatabale, and you shouldn't need to manually fiddle with them like this.

Reply:
The path contains `..` elements, and they are resolved the wrong way. Consider this case:

- a file `/root/a/b'
- a path, to be interpreted relative to the dirname of `b`, which is `../c/d`
- this to resolve to `/root/c/d`
- all is fine

Now consider a symbolic link:
- `/root/s/t/u/v/x/y/z` -> `/root/a/b`
- the dirname of `b` is now `/root/s/t/u/v/x/y`
- the relative path now resolves to `/root/s/t/u/v/x/c/d`
- that file doesn't exist

This is what happens. I don't know where the bug lies – is it the idea that one can use the dirname of a file, or whether symbolic links need to be expanded before doing all this, or whether it's just Meson doing its thing – but without the trickery above things are failing.